fineract-provider/src/main/java/org/apache/fineract/portfolio/client/data/ClientData.java

```diff
@@ -379,7 +383,7 @@ public static ClientData clientIdentifier(final Long id, final String accountNo,
 
         final Long transferToOfficeId = null;
         final String transferToOfficeName = null;
-        final String externalId = null;
+        final ExternalId externalId = null;
```

Comment:
ExternalId type object cannot be allowed to be null! Use `ExternalId.empty()`

Reply:
Done